src/PrettyPrinting.jl

```diff
@@ -2108,8 +2110,57 @@ end
 
 function supercompact(x)
   io = IOBuffer()
-  print(IOContext(io, :supercompact => true), x)
+  print(terse(io), x)
   return String(take!(io))
 end
 
+
+#
+"""
+    terse(io::IO) -> IO
+
+Return a new IO objects derived from `io` for which "supercompact" printing
+mode has been enabled.
+
+See <https://docs.oscar-system.org/stable/DeveloperDocumentation/printing_details/>
+for details.
+
+# Examples
+
+```repl
```

Comment:
`repl` blocks need `@repl` here. But why not even make it a doctest directly?
```suggestion
```@repl
```
```suggestion
```jldoctest
```